### ISSUE TITLE
v4.0 - New annotation system: can't @export specifing the type of the array

### ISSUE THREAD

**Issue by AttackButton** [godotengine/godot#57907]
### Godot version

v4.0.alpha.custom_build [f425d403f]

### System information

Windows 8 and 10

### Issue description

Can't specify the type of the elements using the @export in the new annotation system (https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_exports.html#exporting-arrays) in compassion with the old system.

This is the only way to create arrays right now, I think:
`@export var waypoints: Array = []`

These examples doesn't work:

```
@export(Array, NodePath) var new_paths
@export(Array, PackedScene) var scenes
@export(Array, int) var ints = [1, 2, 3]
```

![image](https://user-images.githubusercontent.com/25422687/153396280-9f4f0f77-59cd-44e8-b59a-edc689a98359.png)

The workaround is to just set the types one by one in the editor.

### Steps to reproduce

Try to create an array with specified type of the elements using the @export.

**Comment by akien-mga** [comment]
Indeed, the above syntax works. The documentation is just outdated (hasn't been ported to 4.0 in that section as you can see from it using `export` and not `@export`. Moving to the docs repo.

2D array exports don't seem to be supported at this time:
```
@export var two_dimensional : Array[Array[float]] = [[1.0, 2.0], [3.0, 4.0]]
```
Returns:
```
Nested typed collections are not supported.
```

It works with untyped arrays (`: Array[Array]`) or:
```
@export var two_dimensional : Array[PackedFloat32Array] = [PackedFloat32Array([1.0, 2.0]), PackedFloat32Array([3.0, 4.0])]
```
